pennylane/_device.py

```diff
@@ -571,6 +573,56 @@ def my_expansion_function(self, tape, max_expansion=10):
         """
         self.custom_expand_fn = types.MethodType(fn, self)
 
+    def _local_tape_expand(self, tape, depth, stop_at):
```

Comment:
Does this need to be a method? Or could it be a standalone helper function. Looks like the only thing on the class it calls is itself.

Reply:
Wait is this just `tape.expand`?

Reply:
I can pull it outside the class. It is tape.expand minus the logic for expanding measurements

Reply:
`Device` is on its last legs, so I'm not too concerned, but I think I'd prefer to have this pulled out as a standalone function.  `Device` has so much stuff already, I don't think this needs to belong to `Device`.